### ISSUE TITLE
End Date/Time doesn't update when only minute of start time changed and dateLimit of 1 minute configured

### ISSUE THREAD

**Issue by manderson23** [dangrossman/daterangepicker#1413]
My picker is configured as follows
```
$('#config-demo').daterangepicker({
  "timePicker": true,
  "timePickerSeconds": true,
  "dateLimit": {
    "minutes": 1
   },
   "locale": {
     "format": "YYYY-MM-DD HH:mm:ss"
    },
    "startDate": moment().startOf('second').subtract(1, 'minutes'),
    "endDate": moment().startOf('second'),
    "autoUpdateInput": true,
    }, function(start, end, label) {
      console.log("New date range selected: ' + start.format('YYYY-MM-DD') + ' to ' + end.format('YYYY-MM-DD') + ' (predefined range: ' + label + ')");
});
```

1. My picker displays as follows:

![step1](https://cloud.githubusercontent.com/assets/1684641/23701461/c3ba04f2-03ee-11e7-8af3-dd84a86926cc.png)

2. I open the picker and select a new minute value for the start time. The time picker combos for the end time update as expected given the dateLimit but the formatted end date/time does not.

![step2](https://cloud.githubusercontent.com/assets/1684641/23701460/c3b9dd92-03ee-11e7-8e37-f6c428c1ff1c.png)

3. I click apply. The start date/time updates as expected but the end date/time is unchanged.

![step3](https://cloud.githubusercontent.com/assets/1684641/23701462/c3ba3d8c-03ee-11e7-8e20-47a257d5b8dc.png)

**Comment by prasanna1211** [comment]
Me too got this issue today. I used react version of this library. Was initially doubting the react version. In my case I used to dynamically feed in time values. I was able to change start and end date dynamically. Also start time was updating dynamically. But end time was not updating. It shows whatever the initial value was there.

**Comment by gangstead** [comment]
I think there is an open PR that fixes this #1410

**Comment by ccipnc** [comment]
`timeChanged` this function has wrong, add the follow code may fix this bug. find `timeChanged: function(e)` this function and fix it like this:

if (isLeft) {
                var endDateLimit = this.endDate.clone();
                var start = this.startDate.clone();
                start.hour(hour);
                start.minute(minute);
                start.second(second);
                this.setStartDate(start);
                if (this.singleDatePicker) {
                    this.endDate = this.startDate.clone();
                } else if (this.endDate && this.endDate.format('YYYY-MM-DD') == start.format('YYYY-MM-DD') && this.endDate.isBefore(start)) {
                    this.setEndDate(start.clone());
                }
                
                if (this.dateLimit && this.startDate.clone().add(this.dateLimit).isBefore(this.endDate)){
                    endDateLimit = this.startDate.clone().add(this.dateLimit);
                    endDateLimit.hour(hour);
                    endDateLimit.minute(minute);
                    endDateLimit.second(second);
                    this.setEndDate(endDateLimit);
                }
            }
